Source/WebCore/Modules/mediasource/MediaSource.cpp

```diff
@@ -697,7 +698,13 @@ ExceptionOr<Ref<SourceBuffer>> MediaSource::addSourceBuffer(const String& type)
         return sourceBufferPrivate.releaseException();
     }
 
-    auto buffer = SourceBuffer::create(sourceBufferPrivate.releaseReturnValue(), this);
+    Ref<SourceBuffer> buffer =
+#if ENABLE(MANAGED_MEDIA_SOURCE)
+        isManaged() ? ManagedSourceBuffer::create(sourceBufferPrivate.releaseReturnValue(), this).get() : SourceBuffer::create(sourceBufferPrivate.releaseReturnValue(), this).get();
```

Comment:
If we want a ManagedMediaSource, I would tend to add the downcast infra

Reply:
I was thinking about that too, good point